design/airgap-existing-cluster.md

```diff
@@ -0,0 +1,110 @@
+- Support different credentials for push / pull (workstation / cluster) to the local registry
+- Support temporary/dynamic and persistent/static credentials for pushing kotsadm images to the registry
+- Support only persistent/static credentials for pulling kotsadm images to the registry
+- Support different credentials for pushing vs pulling images to the local registry
+
+## Non Goals
+
+- Create a single installer for kotsadm + application
+- Create a single download for kotsadm, kots plugin, application and license
+- Automated installs of kotsadm to existing clusters in airgap mode
+- Support private (authenticated) local hosting of kotsadm images
+- Support for installing versions of KOTS different from the KOTS CLI version installed
+
+## Background
+
+It's not currently possible to install KOTS Admin Console into an existing cluster that doesn't have outbound internet access.
+
+## High-Level Design
+
+Airgap Workstation: the computer with kubectl access to the airgap cluster
+Public Workstation: the computer with access to the internet
+
+From the public workstation:
+1. The user downloads the kots plugin suitable for Airgap Workstation.
+1. The user downloads the kotsadm archive containing kots container images
+1. The user downloads the airgap bundle of the application
+1. The user copies the kots plugin, the kotsadm archive and the application airgap bundle onto a medium (eg a USB stick) that can be transfered into the airgapped network.
+
+From the airgap workstation:
+1. The user will install the kots plugin 
+1. The user will copy the container images in the kotsadm archive to the local registry using the kots kubectl plugin
+1. The user will install kotsadm to the cluster using the kots plugin via `kots install` passing local registry information which will be used to generate the kotsadm yaml and image pull secrets (if neccessary)
+1. The user will install thte application and license using the existing airgap installation process 
+
+## Detailed Design
+
+The CI process, on tag, will create the kotsadm archive and include it in the GitHub Release.
+The filename will be `kots_airgap_<version>.tar.gz` and this file contains .tar file exports of all kotsadm images.
+The files SHOULD be stored in the archive in the same format as application airgap files are stored.
+
+Add a `kots admin-console push-images` to the kots cli to retag and push all kotsadm images to an authenticated registry that is accessible from the cluster, using the desired push credentials.
+This new command is used to copy all of the KOTS Admin Console images from one location to another.
+
+Example:
+
+```
+kubectl kots admin-console push-images ./kotsadm-1.16.1.tar.gz registry.somebigbank.com/my-app \
+  --registry-username push-user \
+  --registry-password abcdef
+```
+
+`--registry-username` and `--registry-password` are both optional.
+If not present, attempt to authenticate using the already authenticated local docker credentials.
```

Comment:
As noted above, I would vote this goal be a `Stretch` as well, as docker credential providers can get weird to integrate from what I understand, they involve forking out to other binaries sometimes

Reply:
Good point. I'm fine with making it a stretch.
```suggestion
Stretch: If not present, attempt to authenticate using the already authenticated local docker credentials.
```

Reply:
Although if we don't do that stretch, then what's the behavior without it? Are the credentials required then?

Reply:
When credentials are not provided, we should assume the registry is anonymous.  If it is not, the command will fail with an appropriate error message.

Reply:
@divolgin said on a call that we already have logic for handling trying the registry anonymously and then trying `.docker/config.json` creds if needed, so sounds like this might be easy to do. Again, it can still be a stretch as it's not strictly necessary.